### PR TITLE
ISPN-10280 Deprecate org.infinispan.commons.marshall.Externalizer and…

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.jboss.org/browse/ISPN-10280

### CODE REVIEW

File: commons/src/main/java/org/infinispan/commons/marshall/AdvancedExternalizer.java [hunk 1]
```diff
@@ -41,7 +41,11 @@
  *
  * @author Galder Zamarreño
  * @since 5.0
+ * @deprecated since 10.0, will be removed in a future release. Please configure a {@link
+ * org.infinispan.protostream.SerializationContextInitializer} and utilise ProtoStream annotations on Pojos instead, or
+ * specify a custom marshaller via the SerializationConfiguration.
```
Comment (anistor) [inline commons/src/main/java/org/infinispan/commons/marshall/AdvancedExternalizer.java]: What would be that custom marshaller? Link ?

File: core/src/main/resources/schema/infinispan-config-10.0.xsd [hunk 2]
```diff
@@ -535,8 +535,11 @@
     <xs:sequence>
       <xs:element name="advanced-externalizer" minOccurs="0" maxOccurs="unbounded">
         <xs:annotation>
-          <xs:documentation>
-            AdvancedExternalizer provides an alternative way to provide externalizers for marshalling/unmarshalling user defined classes that overcome the deficiencies of the more user-friendly externalizer definition  model explained in Externalizer.
+          <xs:documentation>Deprecated since 10.0. Please utilise ProtoStream annotations on Pojos instead by
+            configuring a context-initializer, or specifying a custom marshaller for user types.
+            AdvancedExternalizer provides an alternative way to provide externalizers for marshalling/unmarshalling user
```
Comment (anistor) [inline core/src/main/resources/schema/infinispan-config-10.0.xsd]: I don't understand the part after AdvancedExternalizer

Comment (ryanemerson) [reply core/src/main/resources/schema/infinispan-config-10.0.xsd]: I agree, that's just what was there before. I think it was copy and pasted from the AdvancedExternalizer interface javadoc. I've updated to try and give a higher level overview.

File: core/src/main/resources/schema/infinispan-config-10.0.xsd [hunk 3]
```diff
@@ -535,8 +535,12 @@
     <xs:sequence>
       <xs:element name="advanced-externalizer" minOccurs="0" maxOccurs="unbounded">
         <xs:annotation>
-          <xs:documentation>
-            AdvancedExternalizer provides an alternative way to provide externalizers for marshalling/unmarshalling user defined classes that overcome the deficiencies of the more user-friendly externalizer definition  model explained in Externalizer.
+          <xs:documentation>Deprecated since 10.0. Please utilise ProtoStream based marshalling for your Pojos by
```
Comment (anistor) [inline core/src/main/resources/schema/infinispan-config-10.0.xsd]: I'm ok with everything except the word Pojo. Can we just say Java objects?

Comment (ryanemerson) [reply core/src/main/resources/schema/infinispan-config-10.0.xsd]: Updated

**Comment by anistor** [comment]
integrated. thanks @ryanemerson !

**Pull request merged**